compat/src/main/scala-2.11_2.12/scala/collection/compat/CompatImpl.scala

```diff
@@ -12,10 +12,55 @@
 
 package scala.collection.compat
 
+import scala.reflect.ClassTag
 import scala.collection.generic.CanBuildFrom
 import scala.collection.mutable.Builder
 import scala.collection.{immutable => i, mutable => m}
 
+/* builder optimized for a single ++= call, which returns identity on result if possible
+ * and defers to the underlying builder if not.
+ */
+private final class IdentityPreservingBuilder[A, CC[X] <: TraversableOnce[X]](that: Builder[A, CC[A]])(implicit ct: ClassTag[CC[A]])
```

Comment:
Do you mind writing `m.Builder` everywhere instead of `Builder`?

Reply:
done